src/features/student/declaredPrograms/components/interactions/inputs/DeclaredProgramLinkInput/DeclaredProgramLinkInput.test.ts

```diff
@@ -0,0 +1,166 @@
+
+  const stubs = {
+    AvInput: AvInputStub
+  }
+
+  BddTest().when('the component is mounted', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(DeclaredProgramLinkInput, {
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should render the AvInput component', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.exists()).toBe(true)
+    })
+
+    BddTest().then('it should have default label from i18n', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('label')).toBe('Lien')
+    })
+
+    BddTest().then('it should have default placeholder from i18n', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('placeholder')).toBe('https://exemple.com')
+    })
+
+    BddTest().then('it should have default prefix icon', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('prefixIcon')).toBe(MDI_ICONS.LINK)
+    })
+
+    BddTest().then('it should have labelVisible set to true', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('labelVisible')).toBe(true)
+    })
+  })
+
+  BddTest().and('the component is mounted with custom label', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(DeclaredProgramLinkInput, {
+        props: {
+          label: 'Custom Label'
+        },
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should override default label', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('label')).toBe('Custom Label')
+    })
+  })
+
+  BddTest().and('the component is mounted with custom prefix icon', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(DeclaredProgramLinkInput, {
+        props: {
+          prefixIcon: 'mdi:web'
+        },
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should override default prefix icon', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('prefixIcon')).toBe('mdi:web')
+    })
+  })
+
+  BddTest().and('the component is mounted with custom placeholder', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(DeclaredProgramLinkInput, {
+        props: {
+          placeholder: 'Custom Placeholder'
+        },
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should override default placeholder', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('placeholder')).toBe('Custom Placeholder')
+    })
+  })
+
+  BddTest().and('the user enters a URL', () => {
+    beforeEach(async () => {
+      vi.clearAllMocks()
+      wrapper = mount(DeclaredProgramLinkInput, {
+        global: { stubs }
+      })
```

Comment:
```suggestion
  BddTest().and('the user enters a URL', () => {
```

idem que les autres fichiers de test : and dans when et suppression du code répété. devrait être dans ton premier when (component is mounted)